bigquery/vacancy-feedback-metrics-by-month.sql

```diff
@@ -1,83 +1,115 @@
-  CAST(SUM(exclusive_hires_rate_upperbound*vacancies_published) AS INT64) AS exclusive_hires_upperbound,
-  CAST(SUM(exclusivity_rate*vacancies_published) AS INT64) AS exclusive_listings,
-  SUM(exclusive_hires_rate*vacancies_published*IF(category IN ("teacher","leadership"),1200,0)) AS savings, #assume that only teachers and leadership exclusive hires save schools £1200/vacancy, and that other vacancies do not
-  
-  SAFE_DIVIDE(SUM(exclusive_hires),SUM(feedback_available))*SUM(vacancies_published)*1200 AS raw_savings, #estimate of savings, assuming all categories of vacancy behave the same and save schools the same amount of money
-
-  # Extrapolations from hiring staff feedback to estimates of total numbers of vacancies which were hires, exclusive etc, unnormalised (i.e. assuming all categories of vacancy behave the same)
-  SUM(hires_through_tv) AS hires_through_tv_feedback,
-  SUM(hires_through_tv_upperbound) AS hires_through_tv_upperbound_feedback,
-  SUM(exclusive_hires) AS exclusive_hires_feedback,
-  SUM(exclusive_hires_upperbound) AS exclusive_hires_upperbound_feedback,
-  SUM(exclusive_listings) AS exclusive_listings_feedback,
-
-  #Extrapolations from hiring staff feedback to estimates of the overall proportion of vacancies which were hires, exclusive etc, normalised by category
-  SUM(hires_rate_through_tv*vacancies_published)/SUM(vacancies_published) AS hires_rate_through_tv,
-  SUM(hires_rate_through_tv_upperbound*vacancies_published)/SUM(vacancies_published) AS hires_rate_through_tv_upperbound,
-  SUM(exclusive_hires_rate*vacancies_published)/SUM(vacancies_published) AS exclusive_hires_rate,
-  SUM(exclusive_hires_rate_upperbound*vacancies_published)/SUM(vacancies_published) AS exclusive_hires_rate_upperbound,
-  SUM(exclusivity_rate*vacancies_published)/SUM(vacancies_published) AS exclusivity_rate,
-
-  #Extrapolations from hiring staff feedback to estimates of the overall proportion of vacancies which were hires, exclusive etc, unnormalised (i.e. assuming all categories of vacancy behave the same)
-  SAFE_DIVIDE(SUM(hires_through_tv),SUM(feedback_available)) AS raw_hires_rate_through_tv,
-  SAFE_DIVIDE(SUM(hires_through_tv_upperbound),SUM(feedback_available)) AS raw_hires_rate_through_tv_upperbound,
-  SAFE_DIVIDE(SUM(exclusive_hires),SUM(feedback_available)) AS raw_exclusive_hires_rate,
-  SAFE_DIVIDE(SUM(exclusive_hires_upperbound),SUM(feedback_available)) AS raw_exclusive_hires_rate_upperbound,
-  SAFE_DIVIDE(SUM(exclusive_listings),SUM(feedback_available)) AS raw_exclusivity_rate,
-
+  *,
+  #calculate the confidence intervals for these key rates, assuming a 95% confidence level (i.e. z=1.96), that they can be modelled as binomial distributions and that the normal distribution approximation can be applied.
+  1.96*SAFE.SQRT(SAFE_DIVIDE(exclusivity_rate*(1-exclusivity_rate),
+      feedback_available_this_AY_so_far)) AS exclusivity_rate_confidence_interval,
```

Comment:
Is this doing something like making a confidence interval which scales in proportion to (the rate we are interested in), so the shape of the bell curve is squished or stretched out as the rate goes down or up (respectively)? In which case, does feedback_available_this_AY_so_far mean 'the number of feedbacks given so far this year'? If so, then it might be nice to call SUM(feedback_available) `total_feedback_available` so it's a bit clearer what the variable is.

Reply:
@david-mears-dfe - Yes it is, so this is the 'n' you see in the z sqrt(p(1-p)/n) standard formula, where n is the sample size. But I also want to retain the '_this_AY_so_far' etc. because I'm doing totals over that period, over the previous 12 months and over the month (because the data arrives in days, not months). So - I think I'll go with something like 'sample_size_this_AY_so_far'. Does that sound clearer?